src/scrot_selection.c

```diff
@@ -219,8 +219,7 @@ Status scrotSelectionCreateNamedColor(const char *nameColor, XColor *color)
 
 void scrotSelectionGetLineColor(XColor *color)
 {
-    scrotSelectionSetDefaultColorLine();
-
+    assert(opt.lineColor);
     const Status ret = scrotSelectionCreateNamedColor(opt.lineColor, color);
```

Comment:
`scrotSelectionCreateNamedColor()` already `assert()`s that `opt.lineColor` (passed to it as an argument) is not NULL. We're devolving into a situation where the same condition is redundantly checked everywhere. 

Check for invalid values once where they might be acquired, not in callees. Also, does `XAllocNamedColor()` return error for us if we give it null pointers? If so, there would be no point to the `assert()` even for someone who likes it.

Reply:
Mainly added it to be somewhat compatible with the older code. But yeah, I'll remove it.

And even in general, null checks as asserts are not that useful because if something dereferences a NULL pointer - it will typically crash (a loud signal that something's gone wrong).

asserts are more useful for checking invariants which wouldn't otherwise manifest itself loudly (and the program would silently march on forward - moving further away from the actual location where the bug occurred).